azurerm/internal/services/maintenance/data_source_maintenance_configuration.go

```diff
@@ -0,0 +1,78 @@
+package maintenance
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceMaintenanceConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmMaintenanceConfigurationRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Could we add some validation here?

Reply:
I added basic validation for the name: `validation.StringIsNotEmpty`
Because there seems no rule, I tried single `1` as the name, it can succeeded too